syft/frameworks/torch/hook/hook_args.py

```diff
@@ -71,6 +74,7 @@
     AutogradTensor: lambda i: AutogradTensor(data=i).on(i, wrap=False),
     AdditiveSharingTensor: lambda i, **kwargs: AdditiveSharingTensor(**kwargs).on(i, wrap=False),
     MultiPointerTensor: lambda i, **kwargs: MultiPointerTensor(**kwargs).on(i, wrap=False),
+    CRTTensor: lambda i, **kwargs: CRTTensor(i, **kwargs),  # I am not sure how the .on() works here
```

Comment:
Can someone tell me what I did wrong to be unable to use .on() as with other types?

Reply:
.on() allows you to create a empty node before and then to insert it in the syft chain of a tensor, between the head wrapper and the first node. Here you already provide the syft chain as an argument of init so it works in a different way.

Reply:
But (if that's what you meant) `CRTTensor(**kwargs).on(i, wrap=False)` was not working, that's why I chose this. I'll try to see what's wrong with that